File: LeetCode June Challenge/MinimumOperationstoReduceXtoZero.js
var minOperations = function(nums, x) {
    let target = -x, res = -1, sum = 0, left = 0;
    nums.forEach(element => {
        target += element;
    }); 
    if(target === 0) return nums.length;
    if(target < 0) return -1;
    nums.forEach(function(element, index){
        sum += element;
        while(sum > target){
            sum -= nums[left++];
        }
        if(sum === target){
            res = Math.max(res, index - left + 1);
        }
    })
    if(res === -1) return -1;
    else return nums.length - res;
};
var nums = [1,1,4,2,3];
var x = 5;
var ans = minOperations(nums, x);
console.log(ans);

File: LeetCode June Challenge/MinimumOperationstoReduceXtoZero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int target = -x, res = -1, sum = 0, left = 0;
        for(int i : nums) target += i;
        if(target == 0) return nums.length;
        if(target < 0) return -1;
        for(int i = 0; i < nums.length; ++i){
            sum += nums[i];
            while(sum > target){
                sum -= nums[left++];
            }
            if(sum == target){
                res = Math.max(res, i - left + 1);
            }
        }
        return res == -1 ? -1 : nums.length - res;
    }
}
public class MinimumOperationstoReduceXtoZero {
    public static void main(String[] args) {
        int[] nums = {1,1,4,2,3};
        int x = 5;
        Solution s = new Solution();
        System.out.println(s.minOperations(nums, x));
    }
}
